src/api/middleware/validators/company.js

```diff
@@ -71,6 +77,12 @@ export const enable = useExpressValidators([
     existingCompanyParamValidator,
 ]);
 
+export const canBeEditedBy = (req, res, next) => {
+    if (!req.user?.isAdmin && req.user?.companyId !== req.params.companyId)
+        return next(new APIError(HTTPStatus.FORBIDDEN, ErrorTypes.FORBIDDEN, ValidationReasons.UNAUTHORIZED));
+    return next();
+};
```

Comment:
Isn't this already covered by the middleware?

Reply:
Also removed!